source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -111,7 +111,7 @@ bool OsSysCallsImpl::supportsUdpGso() const {
 }
 
 bool OsSysCallsImpl::supportsIpTransparent() const {
-#if !defined(__linux__)
+#if !defined(__linux__) || !defined(IPV6_TRANSPARENT)
```

Comment:
You could probably just remove `!defined(__linux__)`? I don't feel strongly about it either way.